src/Microsoft.AspNet.Mvc.Razor/Compilation/ICompilationService.cs

```diff
@@ -13,11 +11,11 @@ public interface ICompilationService
         /// <summary>
         /// Compiles content and returns the result of compilation.
         /// </summary>
-        /// <param name="fileInfo">The <see cref="IFileInfo"/> for the Razor file that was compiled.</param>
+        /// <param name="fileInfo">The <see cref="RelativeFileInfo"/> for the Razor file that was compiled.</param>
         /// <param name="compilationContent">The generated C# content to be compiled.</param>
         /// <returns>
         /// A <see cref="CompilationResult"/> representing the result of compilation.
         /// </returns>
-        CompilationResult Compile(IFileInfo fileInfo, string compilationContent);
+        CompilationResult Compile(RelativeFileInfo fileInfo, string compilationContent);
```

Comment:
It's part of the work tiem - https://github.com/aspnet/Mvc/issues/955#issuecomment-73305257. We were relying of `PhysicalPath` to create diagnositcs which isn't guaranteed to be available.